file.go

```diff
@@ -125,6 +129,40 @@ func (f *file) TransitiveImports() []File {
 	return out
 }
 
+func (f *file) UnusedImports() []File {
+	public := make(map[int]struct{}, len(f.desc.PublicDependency))
+	for _, i := range f.desc.PublicDependency {
+		public[int(i)] = struct{}{}
+	}
+
+	mp := make(map[string]File, len(f.fileDependencies))
+	for i, fl := range f.fileDependencies {
+		if _, ok := public[i]; ok {
+			continue
+		}
+		mp[fl.Name().String()] = fl
+	}
+
+	for _, msg := range f.AllMessages() {
+		for _, imp := range msg.Imports() {
+			delete(mp, imp.Name().String())
+		}
+	}
+
+	for _, svc := range f.Services() {
+		for _, imp := range svc.Imports() {
```

Comment:
Will `svc.Imports()` include the extensions imported by the service and its methods? E.g. 

```
import "pb/ext/api.proto";
import "pb/ext/data.proto";

service API {
    option (pb.ext.host) = "foobar";
    rpc Do (Request) returns (Response) {
        option (pb.ext.header) = "X-Foo=BAR";
    }
}
```

Reply:
Yep, you'd have to follow the implementation, but walks each method. This is covered in the test as well.